src/main/java/teammates/ui/controller/AdminActivityLogPageAction.java

```diff
@@ -259,16 +258,14 @@ private void generateStatusMessage(List<String> versionToQuery,
             }
             
             ActivityLogEntry activityLogEntry = new ActivityLogEntry(appLog);
-            activityLogEntry = data.filterLogs(activityLogEntry);
-            
-            boolean isToShow = activityLogEntry.toShow() && (!activityLogEntry.isTestingData() || data.getIfShowTestData());
-            if (!isToShow) {
+                       
+            if (!data.filterLogs(activityLogEntry)) {
                 continue;
             }
-            if (isFirstRow) {
-                activityLogEntry.setFirstRow();
-                isFirstRow = false;
+            if (activityLogEntry.isTestingData() && !data.getIfShowTestData()) {
```

Comment:
It's possible to combine it. I am thinking that separating these two may make it more clear.
Like first checking the filter condition, then checking whether is testing data or not.
Nevertheless, I will refactor the log query soon.